### PR TITLE
added login and registration operations

### PR DESCRIPTION
**Pull request by mikhail771**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        }
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<User> query = session.createQuery("from User "
+                    + "where email = :email ", User.class);
+            query.setParameter("email", email);
+
```
Comment (AnaLanda) [inline src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]: You can delete this empty line

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            return Optional.ofNullable(query.getSingleResult());
```
Comment (AnaLanda) [inline src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]:
```suggestion
            return query.uniqueResultOptional();
```

File: src/main/java/com/dev/cinema/model/User.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    @Column(unique = true)
+    private String email;
+    @Column(unique = true)
+    private String login;
+    private String password;
+    private byte[] salt;
+
+    public User() {
+
+    }
```
Comment (AnaLanda) [inline src/main/java/com/dev/cinema/model/User.java]: You can remove this. We'll have a default constructor even if you omit this code)

File: src/main/java/com/dev/cinema/service/impl/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    @Override
+    public User register(String email, String password) {
+        return userService.add(new User(email, password));
+    }
+
+    private boolean isPasswordValid(String password, User user) {
+        return HashUtil.hashPassword(password, user.getSalt())
+                .equals(user.getPassword());
```
Comment (AnaLanda) [inline src/main/java/com/dev/cinema/service/impl/AuthenticationServiceImpl.java]: You can inline this) I think it's under 100 characters long

**Review by AnaLanda — COMMENTED** [review]
Looks good 😃

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java (hunk 1 above)
Comment (mikhail771) [reply src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]: done

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java (hunk 2 above)
Comment (mikhail771) [reply src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]: done

File: src/main/java/com/dev/cinema/model/User.java (hunk 3 above)
Comment (mikhail771) [reply src/main/java/com/dev/cinema/model/User.java:26]: done

File: src/main/java/com/dev/cinema/service/impl/AuthenticationServiceImpl.java (hunk 4 above)
Comment (mikhail771) [reply src/main/java/com/dev/cinema/service/impl/AuthenticationServiceImpl.java]: done

File: src/main/java/com/dev/cinema/model/User.java (hunk 3 above)
Comment (mikhail771) [reply src/main/java/com/dev/cinema/model/User.java]: org.hibernate.InstantiationException: No default constructor for entity:  : com.dev.cinema.model.User

Comment (AnaLanda) [reply src/main/java/com/dev/cinema/model/User.java]: Sorry, you're right! I must have missed that you also have a parameterized constructor 😞

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**